python-backend/app/api/mines/compliance/resources/compliance.py

```diff
@@ -33,33 +33,59 @@ def get(self, mine_no=None):
 
             advisories = 0
             warnings = 0
-            open_orders = 0
-            overdue_orders = 0
+            num_open_orders = 0
+            num_overdue_orders = 0
             section_35_orders = 0
+            open_orders_list = []
 
             for report in data:
-                report_date = datetime.strptime(report.get('assessmentDate'), '%Y-%m-%d %H:%M')
+                report_date = self.get_datetime_from_NRIS_data(report.get('assessmentDate'))
                 one_year_ago = datetime.now() - relativedelta(years=1)
 
                 inspection = report.get('inspection')
                 stops = inspection.get('stops')
+                order_count = 1
 
-                if stops:
-                    stops_dict = stops[0]
+                for stop in stops:
 
-                    stop_orders = stops_dict.get('stopOrders')
-                    stop_advisories = stops_dict.get('stopAdvisories')
-                    stop_warnings = stops_dict.get('stopWarnings')
+                    stop_orders = stop.get('stopOrders')
+                    stop_advisories = stop.get('stopAdvisories')
+                    stop_warnings = stop.get('stopWarnings')
 
-                    if stop_orders:
-                        open_orders += sum(k.get('orderStatus') == 'Open' for k in stop_orders)
-                        overdue_orders += sum(
-                            k.get('orderCompletionDate') is not None
-                            and k.get('orderStatus') == 'Open' and datetime.strptime(
-                                k.get('orderCompletionDate'), '%Y-%m-%d %H:%M') < datetime.now()
-                            for k in stop_orders)
-                        section_35_orders += sum(
-                            k.get('orderAuthoritySection') == 'Section 35' for k in stop_orders)
+                    for order in stop_orders:
+                        if order.get('orderStatus') == 'Open':
+
+                            legislation = order.get('orderLegislations')
+                            permit = order.get('orderPermits')
+                            section = None
+
+                            if legislation:
+                                section = legislation[0].get('section')
```

Comment:
This can still throw `AttributeError: 'NoneType' object has no attribute 'get'. To be safe here, you'll want to check that legislation has an item at index 0

Reply:
I might be wrong but I was under the impression that an empty list evaluates to false in python.

Reply:
Ahh fair enough. I can't say I like using that level of type coercion, as these things get confusing. Especially since our two main languages in this codebase do things differently:
Python
```python
>>> legislation = []
>>> bool(legislation)
False
```

JavaScript
```javascript
> Boolean([])
> true
```

Reply:
I wouldn't consider this resolved. If you're not concerned about this approach to value checking then that's up to you.

But I think a `legislation && len(legislation) > 0` would be more explicit and less reliant on the dynamic type system. Perhaps I'm the only one with this opinion, in which case I will surrender to the majority